test/functional/CacheManagement/src/tests/sharedclasses/options/TestSharedCacheJavaAPI.java

```diff
@@ -124,16 +205,37 @@ public static void main(String args[]) {
 			    				(cacheInfo.getOSsemid() != -1) ||
 			    				(cacheInfo.getLastDetach() == null) ||
 			    				(cacheInfo.getCacheCompressedRefsMode() != compressedRefMode) ||
-			    				(cacheInfo.getCacheLayer() != 0) ||
 			    				(System.currentTimeMillis() - cacheInfo.getLastDetach().getTime() < 0) ||			/* if lastDetach is in future */
 			    				(System.currentTimeMillis() - cacheInfo.getLastDetach().getTime() > (65*60*1000))	/* difference should not be more than 1 hr 5 mins */
 			    			) {
-			    				fail("SharedClassUtilities.getSharedCacheInfo failed for persistent cache: Cache information is not proper");
+			    				fail("SharedClassUtilities.getSharedCacheInfo failed for persistent cache: Cache information is not proper" + 
+			    						"\n cacheName is " + cacheName + 
+			    						"\n cacheInfo.getCacheSize() is " +  cacheInfo.getCacheSize() + 
+			    						"\n cacheInfo.getCacheSoftMaxBytes() is " + cacheInfo.getCacheSoftMaxBytes() +
+			    						"\n cacheInfo.getCacheFreeBytes() is " + cacheInfo.getCacheFreeBytes() +
+			    						"\n cacheInfo.isCacheCompatible() is " + cacheInfo.isCacheCompatible() +
+			    						"\n cacheInfo.isCacheCorrupt() is " + cacheInfo.isCacheCorrupt() +
+			    						"\n cacheInfo.getOSshmid() is " + cacheInfo.getOSshmid() + 
+			    						"\n cacheInfo.getOSsemid() is " + cacheInfo.getLastDetach() + 
+			    						"\n cacheInfo.getCacheCompressedRefsMode() is " + cacheInfo.getCacheCompressedRefsMode() +
+			    						"\n cacheInfo.getCacheLayer() is " + cacheInfo.getCacheLayer() +
+			    						"\n System.currentTimeMillis() - cacheInfo.getLastDetach().getTime() is " + ( System.currentTimeMillis() - cacheInfo.getLastDetach().getTime())) ;
 	  		    			}	
 			    			if ((addrMode.equals("32") && (cacheInfo.getCacheAddressMode() != SharedClassCacheInfo.ADDRESS_MODE_32)) ||
 			    				(addrMode.equals("64") && (cacheInfo.getCacheAddressMode() != SharedClassCacheInfo.ADDRESS_MODE_64))
 			    			) {
-			    				fail("SharedClassUtilities.getSharedCacheInfo failed for persistent cache: Address mode is not proper");
+			    				fail("SharedClassUtilities.getSharedCacheInfo failed for persistent cache: Address mode is not proper " + 
```

Comment:
Print out address mode should be sufficient for this error message.

Reply:
updated.